### PR TITLE
Add Boba Chain Support

### PR DESCRIPTION
**Pull request by AyushBherwani1998**
## Description

Add support for Boba chain. Boba is L2 chain. For more information, [visit the official site](https://boba.network/).

## Testing instructions

Test in same manner as other EVM chains.

## Types of changes
 
New feature (non-breaking change which adds functionality

## Checklist

- [ ] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [ ] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

File: docs/registry.md [hunk 1]
```diff
@@ -75,6 +75,7 @@ This list is generated from [./registry.json](../registry.json)
 | 10000118 | Osmosis          | OSMO   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/osmosis/info/logo.png" width="32" />      | <https://osmosis.zone/>       |
 | 10000145 | Smart Bitcoin Cash | BCH    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/smartbch/info/logo.png" width="32" />     | <https://smartbch.org/>       |
 | 10000250 | Fantom           | FTM    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/fantom/info/logo.png" width="32" />       | <https://fantom.foundation>   |
+| 10000288 | Boba             | BOBAETH | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/boba/info/logo.png" width="32" />         | <https://boba.network/>       |
```
Comment (catenocrypt) [inline docs/registry.md:78]: Logo first needs to be added to assets repository.

**Review by catenocrypt — CHANGES_REQUESTED** [review]

File: docs/registry.md (hunk 1 above)
Comment (AyushBherwani1998) [reply docs/registry.md:78]: Sure, I'll add.

**Comment by catenocrypt** [comment]
Some comments:
- Logo needs to be added to assets repo (blockchain/boba/info, with info.json and logo.png).
- I think some necessary additions to some Kotlin and Swift tests are missing, CI build will show this.  At the minimum, there are some address derivation tests with switch statements where each network must be covered.  Check some other EVM chains.

Otherwise, it looks OK.

File: docs/registry.md (hunk 1 above)
Comment (AyushBherwani1998) [reply docs/registry.md:78]: Submitted a PR: https://github.com/trustwallet/assets/pull/19896

**Comment by AyushBherwani1998** [comment]
>     * I think some necessary additions to some Kotlin and Swift tests are missing, CI build will show this.  At the minimum, there are some address derivation tests with switch statements where each network must be covered.  Check some other EVM chains.

I did check Optimism and Arbitrum PR and it looked fine to me. I have already added the bare minimum switch cases for address derivations for Android and iOS. The checks has also passed, let me know if you still want me to create a separate file for address derivation test under blockchain folder in iOS and Android.

**Review by hewigovens — APPROVED** [review]

File: docs/registry.md (hunk 1 above)
Comment (hewigovens) [reply docs/registry.md:78]: @AyushBherwani1998 can you fix the logo image size?

Comment (AyushBherwani1998) [reply docs/registry.md:78]: @catenocrypt can you check and let me know if it's fine now?

**Review by catenocrypt — APPROVED** [review]

**Pull request merged**